Front-end theme: footer

jefferya opened the pull request:
Addresses #372;

Add the UAL themed footer and associated content attached to the footer - copy from Avalon 5

* add footer and styling
* add content pages and routes for about ERA A+V, Policies, Technologies, Contact, and How to Deposit
* add test to verify content pages exist

murny commented on app/assets/stylesheets/era_custom.scss:
```diff
@@ -1,3 +1,9 @@
+@import 'avalon';
```
Shouldn't need to import avalon here (considering your not using anything from it)? avalon and this file (era_custom) get imported from the application.scss file

murny commented on app/assets/stylesheets/era_custom.scss:
```diff
@@ -1,3 +1,9 @@
+
+$link1: #b5fd99;
+$link1Hover: #efefef;
+$link1Visited: #b5fd99;
```
Sass Variables should be dashed-cased, instead of camelcase

So `$link1Visited` vs `$link-1-visited`.

Also why name this `1`, what does this refer too? Give this a better name? e.g: `$footer-link` vs `$link1`

murny commented on app/views/modules/_footer.html.erb:
```diff
@@ -13,16 +13,55 @@ Unless required by applicable law or agreed to in writing, software distributed
+      <div class="col-sm-4">
+        <h2 class="hidden-xs">Policies</h2>
+        <ul>
+          <li><a href="/about_eraav">About ERA A<span class="plus">&plus;</span>V</a></li>
```
Use rails helpers for these instead of hardcoding links.

`link_to 'About ERA A+V', about_path`

murny commented on app/views/modules/_footer.html.erb:
```diff
@@ -13,16 +13,55 @@ Unless required by applicable law or agreed to in writing, software distributed
+        <h2 class="hidden-xs">Policies</h2>
+        <ul>
+          <li><a href="/about_eraav">About ERA A<span class="plus">&plus;</span>V</a></li>
+          <li><a href="/policies">Policies</a></li>
```
I18n these text/strings as well, like 'Policies'

https://github.com/ualbertalib/avalon/blob/master/config/locales/en.yml

murny commented on app/views/modules/_footer.html.erb:
```diff
@@ -13,16 +13,55 @@ Unless required by applicable law or agreed to in writing, software distributed
+    </div>
+    <div class="row">
+      <a href="http://www.avalonmediasystem.org"><%= t(:powered_by_avalon) %></a>
+      <!-- Avalon Media System Release <%= Avalon::VERSION %> -->
```
why a html comment? This still gets rendered into the HTML source

So put it somewhere better? (meta tag?)

jefferya replied:
@murny do other project capture this info - any suggestions on the name of the meta tag?

murny replied:
There is a Meta generator tag (historically it was used by HTML generators, but has since been used by the software that runs the website. For example any website using WordPress had a Meta generator tag to show the WordPress version. )

Take a peak at this issue in Jupiter that we wish to do (last comment by weiwei where I edited the solution in)

murny commented on app/views/modules/_footer.html.erb:
```diff
@@ -13,16 +13,55 @@ Unless required by applicable law or agreed to in writing, software distributed
+    m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
+    })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');
+
+    ga('create', 'UA-77935097-1', 'auto');
```
This can't be hardcoded like this, needs to be put into secrets.yml or something similar.

(any dev/test/staging/etc environments will sending data to google analytics, heavily skewing your metrics for what you think is production)

murny commented on app/views/pages/deposit.html.erb:
```diff
@@ -0,0 +1,5 @@
+<div class="deposit_page page container">
```
What are some of these classes for like `deposit_page`? There is nothing in the stylesheets and its not being reference elsewhere? Just extra markup that's not needed, therefore should just remove this

murny commented on config/routes.rb:
```diff
@@ -178,6 +178,13 @@
   get '/jobs', to: redirect('/')
 
+  # UofA ERA routes - footer
+  get 'policies' => 'pages#policies', id: 'policies_page'
```
You don't need the `id: 'policies_page'` part? These make the `params[:id]` value to equal `policies_page` which isn't being used anywhere.


Also instead of hash rockets (`=>`) use the form:
```get '/contact', to: 'page#contact'```
instead

murny commented on spec/features/capybara_homepage_spec.rb:
```diff
@@ -18,14 +18,17 @@
   after { Warden.test_reset! }
   it 'validates presence of header and footer on homepage' do
     visit 'http://0.0.0.0:3000'
+    # The theme at U of A doesn't match the upstream theme
```
What does this mean? Does these tests fail because they don't match the upstream theme?

jefferya replied:
Note for when merging upstream changes. Tests modified because the UofA UAL theme differs from the upstream theme and these tests were changed to conform the UAL theme

murny commented on app/assets/stylesheets/era_custom.scss:
```diff
@@ -16,3 +22,46 @@ div.terms-of-use-custom {
+  color: #fff;
+}
+
+.footer {
```
This is fine, but I am wondering if a lot of this custom css is needed? 

For example this footer is in a row-fluid class (which should give the width 100% anyways), can throw a `.text-center` class on it instead of adding it here. 

Font size? Do we really need to play around with this? H2 will be bigger by default, everything else should be fine how it is. (can throw copyright in a `<small>` tag if you want it smaller, etc)

Same with  alot of the unorderlist custom css you have like `.footer ul`. Bootstrap has a style for this already for you! Simply just add a class of `list-unstyled` on the `<ul>` tag.

jefferya replied:
Footer CCS (and most code) pulled directly from UAL Avalon 5. I did not take the time to rationalize to help in finishing closer to the deadline. 

I agree with @murny  point in the review.

murny commented on app/views/pages/about_eraav.html.erb:
```diff
@@ -0,0 +1,14 @@
+<div class="about_page page container">
+	<h2>About ERA A<span class="plus">&plus;</span>V</h2>
```
Got some weird indentation/spacing here? It's like 4+ spaces? 

Also some html files have ending newlines (or multiple) and others don't

jefferya commented on app/controllers/pages_controller.rb:
```diff
@@ -0,0 +1,21 @@
+class PagesController < ApplicationController
+
+  def about_eraav
```
Existing route for `about`. Leads to the Avalon 6 stack` about/health` check page

jefferya commented:
Footer CCS (and most code) pulled directly from UAL Avalon 5. I did not take the time to rationalize to help in finishing closer to the deadline and had assumed the code had already passed a review. 

I agree with @murny regarding the comments in the review.

murny commented:
@jefferya We just had a conversation about this (Peter, Weiwei and myself). Think the plan going forward was to merge these 3 PRs how they are (with one fix, which is to make the 3 videos for the homepage as static images instead). And then maybe document some of these suggestions as future issues.

weiweishi commented:
Thanks @murny for the summary. @jefferya if there's any really low hanging fruit such as rename the class name, or separating era-custom.scss to footer/header.scss, that you feel you'd be able to tackle fairly easily, please feel free to make those changes. I will capture whatever is left behind into an issue once you have made the changes you are available to make, and we will try to work on these improvement after the Avalon upgrade is completed. Also opened a PR on DI_internal to keep track tips/guides we'd need to follow for when developing/reviewing html/css related changes.

weiweishi commented:
Thank you @jefferya and @murny for your patience, and apology for your frustrations along the way, while we continue to work on the gaps and areas need improvement in our process and in our work. We all have learned more from this! Your work and suggestions are truly appreciated!

jefferya commented:
@murny I've refactored the code as per the suggestions. 

Changes address:
- CSS fixes removing extra imports, refactoring colour as variables; dashed-cased variables; footer CSS in footer-specific file; leverage more Bootstrap functionality and remove CSS
- removed unnecessary controller
- static content pages: removed unnecessary div; fixed tab style; end-of-file style; removed unused CSS classes 
- links: leveraged Rails link_to helpers
- routes: use `gt '/contact', to: 'page#contact'` style as opposed to hash rockets ( => )
- refactored to use more i18n  
- google analytics code: leverage secrets.yml
